hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/UpsertTargetUtils.java

```diff
@@ -322,10 +322,14 @@ private static void setCompactFields(
                     recordBuilder.setTimestampWithTimezone(field.getName(), (OffsetDateTime) fieldValue);
                     break;
                 case OBJECT:
-                    final GenericRecordBuilder nestedRecordBuilder = GenericRecordBuilder
-                            .compact(field.getDataType().getObjectTypeMetadata());
-                    setCompactFields((RowValue) fieldValue, field.getDataType(), nestedRecordBuilder);
-                    recordBuilder.setGenericRecord(field.getName(), nestedRecordBuilder.build());
+                    if (fieldValue == null) {
+                        recordBuilder.setGenericRecord(field.getName(), null);
```

Comment:
`GenericRecordBuilder` doesn't allow setting the same field twice. To ensure this, `DeserializedGenericRecordBuilder` and `DeserializedSchemaBoundGenericRecordBuilder` check if `Map.putIfAbsent(name, value) != null`, which actually allows overwrites as long as the previous value is null. On the contrary, `DeserializedGenericRecordCloner` and `PortableGenericRecordBuilder` keep a set of field names and a boolean array respectively, which properly guard against overwrites. Doesn't this inconsistency need to be eliminated?

Reply:
Actually checked and it won't cause Compact schema duplication since fingerprint value doesn't actually include sub-field schema fingerprint, in fact the schema of child object is completely opaque for the parent object schema.